### ISSUE TITLE
Could not catch children after updateChild

### ISSUE THREAD

**Issue by monolithed** [antvis/G6#3477]
### Describe the bug

I wanted to add a collapse icon to nodes that have children, but it does not work with dynamic nodes. 
The problem is that I couldn't catch the children property from the cfg object.

### Your Example Website or App

```js
if (cfg.children) {
     group.addShape("marker", {
          attrs: {
              x: 60 - w / 2,
              y: 40 - h / 2,
              r: 6,
              stroke: "#ff4d4f",
              cursor: "pointer",
              symbol: cfg.collapsed ? G6.Marker.expand : G6.Marker.collapse
          },
          name: "collapse-item"
     });
}
```

```js
graph.on("node:click", (evt) => {
  const { item, target } = evt;
  const targetType = target.get("type");
  const name = target.get("name");

  if (targetType === "marker") {
    const model = item.getModel();

    if (name === "add-item") {
      if (!model.children) {
        model.children = [];
      }

      const id = `n-${Math.random()}`;
      model.children.push({id});
      graph.updateChild(model, model.id);
    }
    else if (name === "remove-item") {
      graph.removeChild(model.id);
    }
  }
});
```

https://codesandbox.io/s/autumn-snowflake-j9lof?file=/index.js

### Steps to Reproduce the Bug or Issue

Click add a new node

### Expected behavior

The collapse icon is present

### Screenshots or Videos

https://user-images.githubusercontent.com/803674/150701057-f6da7af7-3e60-425d-b98d-a706a4e7d789.mov

### Platform

- OS: macOS
- Browser: Safari
- Version: latest

**Comment by Yanyan-Wang** [comment]
replace `model.collapsed = !model.collapsed;` to:

graph.updateItem(item, {
    collapsed: !model.collapsed
  })

**Comment by monolithed** [comment]
@Yanyan-Wang thanks for your reply, but the cfg.children property is still not available on new nodes to show or hide the collapse icon

**Comment by Yanyan-Wang** [comment]
There were some problems about TreeGraph API: updateChild, updateChildren, addChid, removeChild did not refresh the item after data changed. The problem is solved by v4.8.0.

But, there are some problems in your demo
1. updateChild is not used properly. See docs here  http://g6.antv.antgroup.com/en/api/treemethods#updatechilddata-parentid
![image](https://user-images.githubusercontent.com/29593318/207852882-ff3fb236-b53e-400d-b647-909abf46f261.png)
But you call it `graph.updateChild(model, model.id)` when click the + icon, which means take model as the child into model itself. I think the proper way to add child to this model is:
```
const children = model.children || [];
const id = `n-${Math.random()}`;
const newChildData = {
      id,
      label: id.substr(0, 8)
};
children.push(newChildData);
graph.updateChildren(children, model.id);
```
Where `updateChildren` is called instead of `updateChild`. Notice the differences between these two API:

![image](https://user-images.githubusercontent.com/29593318/207853618-e36e9022-ac64-4f52-89e1-99e9bc297fe7.png)


2. You called `graph.removeChild(model.id)`. Did you want to remove the node itself? If so, the way you implemented is correct. And the root node is not one's child. If you wanted to remove the children of the clicked node, call `graph.removeChildren([], model.id)` instead.


Here goes the correct demo: https://codesandbox.io/s/polished-bird-644t9v?file=/index.js
